corehq/apps/users/views/mobile/users.py

```diff
@@ -1685,6 +1731,60 @@ def link_connectid_user(request, domain):
         return HttpResponse()
 
 
+@require_can_edit_commcare_users
+@location_safe
+def send_connectid_invite(request, domain, user_id):
+    # Currently same as what send_confirmation_sms does
+    user = CommCareUser.get_by_user_id(user_id, domain)
+    if user.is_account_confirmed or not user.is_commcare_user() or user.domain != domain:
+        messages.error(request, "The user is already confirmed or is not a mobile user")
+        return HttpResponse(status=400)
+
+    invite_code = encrypt_account_confirmation_info(user)
+    url = absolute_reverse("confirm_connectid_user", args=[user.domain])
+    text_content = f"""
+    Link your ConnectID account to CommCare
+    username: {user.username}
+    domain: {domain}
+    invite_code: {invite_code}
+    url: {url}
+    Make a POST call to above url with invite_code and your ConnectID token
+    """
+    return send_sms(
+        domain=domain,
+        contact=None,
+        phone_number=user.default_phone_number,
+        text=text_content)
+
+
+@csrf_exempt
+@require_POST
+def confirm_connectid_user(request, domain):
+    try:
+        token = request.POST["token"]
+        invite_code = request.POST["invite_code"]
+        invite = json.loads(b64_aes_decrypt(invite_code))
+        user_id = invite['user_id']
+        expiry_time = invite['time']
+    except ValueError:
+        return HttpResponseBadRequest("Invalid Request")
+
+    # Expiry limit is 3 days
+    if float(int(time.time()) - expiry_time) > 3 * 24 * 60 * 60:
+        return JsonResponse(data={"error": "Invite has expired"})
+
+    user = CommCareUser.get(user_id)
+    if user.domain != domain:
+        return HttpResponseBadRequest("Invalid Request")
+
+    connectid_username = get_connectid_userinfo(token)
```

Comment:
Does this confirm that the token was valid at all? What response do we send if it wasn't?

Reply:
If the token is not valid this would not return a username. So right now, this would 500 if the token is not valid. Which is same behaviour as that of [link_connectid_user](https://github.com/dimagi/commcare-hq/pull/34967/files#diff-351c05060f5e3368f3b74b23ed62403574b23ed3c447122980009471e2833648R1720).